### ISSUE TITLE
Self hosting zeolite

### ISSUE THREAD

**Issue by samouri** [ta0kira/zeolite#153]
Hi @ta0kira! 

Have you thought about [self hosting](https://en.wikipedia.org/wiki/Self-hosting_(compilers)) zeolite?

**Comment by ta0kira** [comment]
Thanks for the suggestion. I've considered it, but it seems impractical without doing a lot of design work to make Zeolite less painful when it comes to processing text. I also haven't decided if I want to get into exceptions or monad semantics for statement-level short-circuiting.

As far as project direction, I don't foresee eliminating [`clang++`](https://clang.llvm.org/) as the compiler backend without a whole lot more people working on the project. AFAIK, even migrating to [LLVM](https://llvm.org/) would still require dealing with the linker on the host OS, which is actually a step in the wrong direction. Either way, I think it's critical to support simple language extensions via C++. (I'm using Python as a model for success here, at least from the perspective of versatility.)

In summary, offloading the heavy lifting to C++ and Haskell lets me spend more time thinking about what I'd like the language to do, rather than the endless portability considerations that other languages have already solved.
